packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1906,6 +1910,11 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
         return SelectionResult.end;
       }
     }
+    if (lastSelectionResult == SelectionResult.next) {
+      return SelectionResult.next;
+    } else if (lastSelectionResult == SelectionResult.previous) {
+      return SelectionResult.previous;
```

Comment:
This doesn't look right. consider these cases, assuming 5 child

1.
selectable 1 -> skipped (continue)
selectable 2 -> next  (continue)
selectable 3 -> next  (continue)
selectable 4 -> prev  (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 3 or 4
should return SelectionResult.end

2.
selectable 1 -> skipped (continue)
selectable 2 -> prev  (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end

3.
selectable 1 -> prev  (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 1
should return SelectionResult.prev

4.
selectable 1 -> skip  (continue)
selectable 2 -> next  (continue)
selectable 3 -> end  (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 3
should return SelectionResult.end

5.
selectable 1 -> skip  (continue)
selectable 2 -> next  (continue)
selectable 3 -> skip  (stop here)

currentSelectionStartIndex/currentSelectionEndIndex : 2
should return SelectionResult.end

6.
selectable 1 -> skip  (continue)
selectable 2 -> skip  (continue)
selectable 3 -> skip  (continue)
selectable 4 -> next  (continue)
selectable 5 -> next  (at the end)

currentSelectionStartIndex/currentSelectionEndIndex : 5
should return SelectionResult.next